DecipherNow/helm-charts
# JWT filter integration

## Changes
- Added egress listeners and domains for all proxies
- Added an egress route `/jwt` for all proxies
- Added the jwt security filter to all ingress listeners (except jwt-security listener), and the inheader filters to all egress listeners
- Updated control plane services to latest versions

## Testing
- Make sure you have [installed helm 3](https://helm.sh/docs/intro/install/)
- If you don't have credentials.yaml filled out, run `make credentials`
- Run `install.sh` script in the root directory which will install the mesh - this is a temporary script while @moosecanswim works on the makefile. Note that the installation will take at least 3min, I usually run `watch kubectl get pods`. Also, if for some reason you see a voyager/edge related resource error, even if you see the edge pod, uninstall and reinstall it `helm uninstall edge && helm install edge edge --set=global.environment=kubernetes`
- Make sure you can hit the dashboard at `$(minikube ip):30000`
- Check out sidecar logs for edge and see the userpolicy jwt token header is set
